scripts/build/build_examples.py

```diff
@@ -59,6 +59,18 @@ def ValidateRepoPath(context, parameter, value):
     return value
 
 
+def ValidateTargetNames(context, parameter, values):
```

Comment:
This will double-loop to try every target for every possible target.

We already likely attempt matches, so maybe we should throw at:
https://github.com/project-chip/connectedhomeip/blob/master/scripts/build/build/__init__.py#L45

```py
#...
        for target in targets:
            found = False
            for choice in BUILD_TARGETS:
                builder = choice.Create(target, self.runner, self.repository_path,
                                        self.output_prefix, options)
                if builder:
                    self.builders.append(builder)
                    found = True

            if not found:
                logging.error(f"Target '{target}' could not be found. Nothing executed for it")
                # Should we raise an exception here instead of just logging?
# ...
```

Reply:
I was also thinking about returning a "Target" object from the validator. The target might be a name + the output of `StringIntoTargetParts`. Because right now there is also duplication of `str.lower()`. The performance gain will be infinitesimal (single `.lower()` and not duplicated loop), but the flow will be clearer.